lib/liquid/tags/include.rb

```diff
@@ -45,7 +44,13 @@ def parse(tokens)
     def render(context)
       partial = load_cached_partial(context)
       variable = context.evaluate(@variable_name)
-
+      
+      if @template_name =~ QuotedString
+          context_variable_name = @template_name[1..-2].split('/'.freeze).last
+      else
+          context_variable_name = context[@template_name]
+      end
```

Comment:
This should stay in `initialize`, otherwise you will repeat this parsing work on every `render` call (although it's result will never change).